[Woo POS][Refunds] Add "Review refund" step in POS refund flow

samiuelson opened the pull request:
WOOMOB-1826

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR introduces 2nd step in the POS refund flow. The "Review refund" step is shown after clicking "Continue" button in the item selection step. The changes include state management enhancements with a new step enum, formatted price fields for currency display, and test coverage for step navigation.

What's not included:
* Refund reason - it's hardcoded for now, though the text style should match design specs
* Edit reason button - it's not working yet
* Via (payment method) - hardcoded text for now

<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

### Test Steps
<!-- Describe how to test your changes. Include only what’s needed for the reviewer to validate the behavior.
For new features, outline the main user flow or goal rather than every tap or click — the reviewer should be able to complete the flow naturally.
If applicable, mention key devices, scenarios, or edge cases to verify. -->
1. Go to Woo POS > Orders
2. Tap "Issue refund" button and explore the refund flow 

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
<img width="800" src="https://github.com/user-attachments/assets/1431f0c6-d17c-47ab-90da-098a58b4ee92" />

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Implement "review refund" step in POS refund flow

- Introduce `RefundStep` (SelectItems, ReviewRefund) in `WooPosRefundState` to manage the refund flow.
- Add formatted price fields to `WooPosRefundState.Content` and populate them in `WooPosRefundViewModel` using `CurrencyFormatter`.
- Update `WooPosIssueRefundDialog` to render either item selection or review content based on the current step.
- Rename `RefundDialogContent` to `SelectItemsContent`.
- Add `ReviewRefundContent` composable to display refund summary, including subtotal, taxes, and total.
- Add `onContinueToReview` and `onBackToSelectItems` navigation logic to `WooPosRefundViewModel`.
- Add string resources for the refund review screen.

samiuelson committed:
Change refund dialog height to "wrap-content"

samiuelson committed:
Refactor refund dialog to use UI events

samiuelson committed:
Reset refund step when the dialog is dismissed

samiuelson committed:
Add tests

samiuelson committed:
Keep mocked strings away from res

samiuelson committed:
Recreate WooPosRefundViewModel on each selected order_id change

samiuelson committed:
Use correct plurals string

samiuelson commented on WooCommerce/src/main/res/values/strings.xml:
```diff
@@ -3803,6 +3803,21 @@
+    <string name="woopos_orders_refund_for_order">Refund for %1$s</string>
+    <string name="woopos_orders_issue_refund">Issue Refund</string>
+    <string name="woopos_orders_items_subtotal_count">Items subtotal (%1$d item)</string>
+    <plurals name="woopos_orders_items_subtotal_count_plural">
```
Updated: [fbfaeee](https://github.com/woocommerce/woocommerce-android/pull/15064/commits/fbfaeee847a880d9515c8fbc9e6381108507e039)

samiuelson committed:
Merge branch 'trunk' into woomob-1826-show-review-refund-step

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosIssueRefundDialog.kt:
```diff
@@ -159,12 +179,12 @@ private fun NoItemsContent(onDismissRequest: () -> Unit) {
     onContinue: () -> Unit
 ) {
-    Column(modifier = Modifier.fillMaxSize()) {
+    Column(modifier = Modifier.fillMaxWidth()) {
```
💡 This makes the dialog adjust its height to the content rather than taking the whole height of the screen.

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosIssueRefundDialog.kt:
```diff
@@ -336,6 +356,185 @@ private fun RefundableItemRow(item: WooPosRefundableItem) {
+                    isTotal = true
+                )
+                WooPosText(
+                    text = "Via payment card ••••1456",
```
This will be replaced with real value in the next iteration.

malinajirka replied:
🔍 In general, I think it's safer to put an apparently wrong text for testing - just to ensure we don't miss it. E.g. "TEST: Via payment card ••••9999".

samiuelson replied:
Updated: [375c96f](https://github.com/woocommerce/woocommerce-android/pull/15064/commits/375c96fc5dfeb5861ad8f03f25070d7a98bc183c)

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosIssueRefundDialog.kt:
```diff
@@ -336,6 +356,185 @@ private fun RefundableItemRow(item: WooPosRefundableItem) {
+                    )
+                }
+                WooPosText(
+                    text = "Customer bought an extra item.",
```
This as well.

samiuelson committed:
Clean up code

samiuelson committed:
Clean up code

samiuelson committed:
Merge remote-tracking branch 'origin/woomob-1826-show-review-refund-step' into woomob-1826-show-review-refund-step

samiuelson committed:
Remove unused res

samiuelson commented on WooCommerce/src/main/res/values/strings.xml:
```diff
@@ -3803,6 +3803,18 @@
     <string name="woopos_orders_items_selected_count">(%1$d SELECTED)</string>
     <string name="woopos_orders_loading_refund_items">Loading refundable items</string>
     <string name="woopos_orders_no_items_available_for_refund">No items available for refund</string>
+    <string name="woopos_orders_review_refund_content_description">Review refund details</string>
```
Removed: [09ff917](https://github.com/woocommerce/woocommerce-android/pull/15064/commits/09ff917275c7eb53ef9ebc9c311d88f295ff7eaa)

samiuelson commented on WooCommerce/src/main/res/values/strings.xml:
```diff
@@ -3803,6 +3803,18 @@
     <string name="woopos_orders_no_items_available_for_refund">No items available for refund</string>
+    <string name="woopos_orders_review_refund_content_description">Review refund details</string>
+    <string name="woopos_orders_review_refund">Review refund</string>
+    <string name="woopos_orders_refund_for_order">Refund for %1$s</string>
```
Removed: [09ff917](https://github.com/woocommerce/woocommerce-android/pull/15064/commits/09ff917275c7eb53ef9ebc9c311d88f295ff7eaa)

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosIssueRefundDialog.kt:
```diff
@@ -52,11 +54,10 @@ import java.math.BigDecimal
 ) {
     val viewModel: WooPosRefundViewModel =
-        hiltViewModel<WooPosRefundViewModel, WooPosRefundViewModel.Factory> { factory ->
+        hiltViewModel<WooPosRefundViewModel, WooPosRefundViewModel.Factory>(key = "refund_$orderId") { factory ->
```
💡 This recreates the VM when another order gets selected in the Orders pane.

malinajirka reviewed (approved):
Thanks @samiuelson, great job! I've found a couple UI issues - they are not all related to the PR, but I thought it's better to mention them.


1. 
<img width="621" height="580" alt="image" src="https://github.com/user-attachments/assets/f3026528-c295-4044-aa5f-1a1f195736f8" />

2. I noticed the Refund dialog appears instantly without an animation - feels a bit off and inconsistent.

3. Semi-related to the above - moving between steps is also without an animation.

4. If I proceed to the `Review refund` step and I dismiss the dialog, upon reopening the flow, I'm on the review refund step instead of item selection step.

samiuelson committed:
Clean up dividers

samiuelson committed:
Reset refund step after last "Continue" tap

samiuelson committed:
Update texts

samiuelson commented:
Thanks for review!

> 1. Dividers inconsistency

Fixed: abe5ee4

> 2. I noticed the Refund dialog appears instantly without an animation - feels a bit off and inconsistent.
> 3. Semi-related to the above - moving between steps is also without an animation.

I created task to add animations and will work on it after in next iterations: WOOMOB-1861

> 4. If I proceed to the `Review refund` step and I dismiss the dialog, upon reopening the flow, I'm on the review refund step instead of item selection step.

Fixed: 102c699

samiuelson committed:
Merge branch 'trunk' into woomob-1826-show-review-refund-step

samiuelson committed:
Satisfy detekt's complaints

samiuelson committed:
Merge remote-tracking branch 'origin/woomob-1826-show-review-refund-step' into woomob-1826-show-review-refund-step
